packages/amplify-graphql-model-transformer/src/resources/custom-resource-lambda/custom-resource-handler.ts

```diff
@@ -0,0 +1,440 @@
+      IndexName: gsiToAdd.IndexName,
+      KeySchema: gsiToAdd.KeySchema,
+      Projection: gsiToAdd.Projection,
+      ProvisionedThroughput: gsiToAdd.ProvisionedThroughput,
+    }) as DynamoDB.CreateGlobalSecondaryIndexAction;
+    return {
+      TableName: currentState.TableName!,
+      AttributeDefinitions: endState.AttributeDefinitions.filter((def) => attributeNamesToInclude.includes(def.AttributeName)),
+      GlobalSecondaryIndexUpdates: [
+        {
+          Create: gsiToAddAction,
+        },
+      ],
+    };
+  }
+
+  // no more updates necessary
+  return undefined;
+};
+
+type CreateTableResponse = {
+  tableName: string;
+  tableArn: string;
+  streamArn?: string;
+};
+
+/**
+ * Extract the custom DynamoDB table properties from event
+ * @param event Event for onEvent or isComplete
+ * @returns The table input for Custom dynamoDB Table
+ */
+const extractTableInputFromEvent = (
+  event: AWSCDKAsyncCustomResource.OnEventRequest | AWSCDKAsyncCustomResource.IsCompleteRequest,
+): CustomDDB.Input => {
+  // isolate the resource properties from the event and remove the service token
+  const resourceProperties = { ...event.ResourceProperties } as Record<string, any> & { ServiceToken?: string };
+  delete resourceProperties.ServiceToken;
+
+  // cast the remaining resource properties to the DynamoDB API call input type
+  const tableDef = convertStringToBooleanOrNumber(resourceProperties) as CustomDDB.Input;
+  return tableDef;
+};
+/**
+ * Util function to convert string values to the correct form
+ * Such as 'true' to true, '5' to 5
+ * @param obj Input object
+ * @returns Oject with its values converted to the correct form of boolean or number
+ */
+const convertStringToBooleanOrNumber = (obj: Record<string, any>): Record<string, any> => {
+  for (const key in obj) {
+    if (typeof obj[key] === 'object') {
+      // If the property is an object, recursively call the function
+      obj[key] = convertStringToBooleanOrNumber(obj[key]);
+    } else if (typeof obj[key] === 'string') {
+      if (obj[key] === 'true' || obj[key] === 'false') {
+        // If the property is a string with value 'true' or 'false', convert it to a boolean
+        obj[key] = obj[key] === 'true';
+      } else if (!isNaN(Number(obj[key]))) {
+        // If the property is a string that can be parsed into a number, convert it to a number
+        obj[key] = Number(obj[key]);
+      }
+    }
+  }
+  return obj;
+};
+/**
+ * Util function to remove undefined values from root level of object
+ * @param obj Input Object
+ * @returns obj without undefined attributes
+ */
+const removeUndefinedAttributes = (obj: Record<string, any>): Record<string, any> => {
+  for (const key in obj) {
+    if (obj[key] === undefined) {
+      // Use the delete operator to remove the attribute if it's undefined
+      delete obj[key];
+    }
+  }
+  return obj;
+};
+
+const toCreateTableInput = (props: any): CreateTableInput => {
```

Comment:
Mark this as TODO. This will be addressed by the latter PR where the `CustomDDB.Input` is well defined. The input will be different from current as the new table construct is in use. (There are some additional property parse process such as using PascalCase, converting table props to CFN properties, etc)